test/e2e/settings/pod_security_policy.go

```diff
@@ -0,0 +1,149 @@
+	corev1 "k8s.io/api/core/v1"
+	extensions "k8s.io/api/extensions/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+const (
+	ingressControllerPSP = "ingress-controller-psp"
+)
+
+var _ = framework.IngressNginxDescribe("Pod Security Policies", func() {
+	f := framework.NewDefaultFramework("pod-security-policies")
+
+	BeforeEach(func() {
+		psp := createPodSecurityPolicy()
+		_, err := f.KubeClientSet.Extensions().PodSecurityPolicies().Create(psp)
+		if !k8sErrors.IsAlreadyExists(err) {
+			Expect(err).NotTo(HaveOccurred(), "creating Pod Security Policy")
+		}
+
+		role, err := f.KubeClientSet.RbacV1().ClusterRoles().Get("nginx-ingress-clusterrole", metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred(), "getting ingress controller cluster role")
+		Expect(role).NotTo(BeNil())
+
+		role.Rules = append(role.Rules, rbacv1.PolicyRule{
+			APIGroups:     []string{"policy"},
+			Resources:     []string{"podsecuritypolicies"},
+			ResourceNames: []string{ingressControllerPSP},
+			Verbs:         []string{"use"},
+		})
+
+		_, err = f.KubeClientSet.RbacV1().ClusterRoles().Update(role)
+		Expect(err).NotTo(HaveOccurred(), "updating ingress controller cluster role to use a pod security policy")
+
+		// update the deployment just to trigger a rolling update and the use of the security policy
+		err = framework.UpdateDeployment(f.KubeClientSet, f.IngressController.Namespace, "nginx-ingress-controller", 1,
+			func(deployment *appsv1beta1.Deployment) error {
+				args := deployment.Spec.Template.Spec.Containers[0].Args
+				args = append(args, "--v=2")
+				deployment.Spec.Template.Spec.Containers[0].Args = args
+				_, err := f.KubeClientSet.AppsV1beta1().Deployments(f.IngressController.Namespace).Update(deployment)
+
+				return err
+			})
+		Expect(err).NotTo(HaveOccurred())
+
+		f.NewEchoDeployment()
+	})
+
+	// running tests in parallel can update the cluster roles, which introduce a failure
+	/*
+		AfterEach(func() {
+			role, err := f.KubeClientSet.RbacV1().ClusterRoles().Get("nginx-ingress-clusterrole", metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "getting ingress controller cluster role")
+			Expect(role).NotTo(BeNil())
+
+			index := -1
+			for idx, rule := range role.Rules {
+				found := false
+				for _, rn := range rule.ResourceNames {
+					if rn == ingressControllerPSP {
+						found = true
+						break
+					}
+				}
+				if found {
+					index = idx
+				}
+			}
+
+			role.Rules = append(role.Rules[:index], role.Rules[index+1:]...)
+			_, err = f.KubeClientSet.RbacV1().ClusterRoles().Update(role)
+			Expect(err).NotTo(HaveOccurred(), "updating ingress controller cluster role to use a pod security policy")
+		})
+	*/
+
+	It("should be running with a Pod Security Policy", func() {
+		f.WaitForNginxConfiguration(
+			func(cfg string) bool {
+				return strings.Contains(cfg, "server_tokens on")
```

Comment:
does `server_tokens` matter? or is this here to make sure nginx configuration is generated? maybe send a request and assert 404 instead?